datafusion/substrait/src/variation_const.rs

```diff
@@ -37,3 +38,58 @@ pub const DEFAULT_CONTAINER_TYPE_REF: u32 = 0;
 pub const LARGE_CONTAINER_TYPE_REF: u32 = 1;
 pub const DECIMAL_128_TYPE_REF: u32 = 0;
 pub const DECIMAL_256_TYPE_REF: u32 = 1;
+
+// For custom types
+/// For [`DataType::Interval`] with [`IntervalUnit::YearMonth`].
+///
+/// An `i32` for elapsed whole months. See also [`ScalarValue::IntervalYearMonth`]
+/// for the literal definition in DataFusion.
+///
+/// [`DataType::Interval`]: datafusion::arrow::datatypes::DataType::Interval
+/// [`IntervalUnit::YearMonth`]: datafusion::arrow::datatypes::IntervalUnit::YearMonth
+/// [`ScalarValue::IntervalYearMonth`]: datafusion::common::ScalarValue::IntervalYearMonth
+pub const INTERVAL_YEAR_MONTH_TYPE_REF: u32 = 1;
+
+/// For [`DataType::Interval`] with [`IntervalUnit::DayTime`].
+///
+/// An `i64` as:
+/// - days: `i32`
+/// - milliseconds: `i32`
+///
+/// See also [`ScalarValue::IntervalDayTime`] for the literal definition in DataFusion.
+///
+/// [`DataType::Interval`]: datafusion::arrow::datatypes::DataType::Interval
+/// [`IntervalUnit::DayTime`]: datafusion::arrow::datatypes::IntervalUnit::DayTime
+/// [`ScalarValue::IntervalDayTime`]: datafusion::common::ScalarValue::IntervalDayTime
+pub const INTERVAL_DAY_TIME_TYPE_REF: u32 = 2;
+
+/// For [`DataType::Interval`] with [`IntervalUnit::MonthDayNano`].
+///
+/// An `i128` as:
+/// - months: `i32`
+/// - days: `i32`
+/// - nanoseconds: `i64`
+///
+/// See also [`ScalarValue::IntervalMonthDayNano`] for the literal definition in DataFusion.
```

Comment:
FYI the interval implementation is changing in the next arrow I think: https://github.com/apache/arrow-rs/pull/5769

Reply:
Thanks for informing, I'd like to help migrate to the new arrow version. BTW, is there any plan for next bump?

Reply:
I think ETA is next week sometime https://github.com/apache/arrow-rs/issues/5688

Maybe you can make a "pre-release" of DataFusion against the un-released version of arrow-rs (which @tustvold  often does to make sure as a way to sanity check the release)

Reply:
Good :+1: The "pre-update" is https://github.com/apache/datafusion/pull/10765 (still WIP